scripts/test_pr_convergence.py

```
@@ -0,0 +1,838 @@
+            path.write_text("{bad", encoding="utf-8")
+            with self.assertRaises(pr_convergence.DataSourceError) as raised:
+                pr_convergence.load_snapshot(path)
+
+        self.assertEqual(raised.exception.source, "snapshot")
+
+    def test_group_accessible_snapshot_is_refused(self) -> None:
+        with tempfile.TemporaryDirectory() as tmpdir:
+            path = pathlib.Path(tmpdir) / "snapshot.json"
+            path.write_text("{}", encoding="utf-8")
+            path.chmod(0o640)
+            with self.assertRaises(pr_convergence.DataSourceError) as raised:
```

Ruff (0.16.0)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)